source/BonusList.cpp

```diff
@@ -46,36 +46,57 @@ namespace Duel6 {
     }
 
     void BonusList::addRandomBonus() {
-        const Level &level = world.getLevel();
         bool weapon = (Math::random(2) == 1);
-        Int32 x, y, attempts = 0;
-
-        do {
-            attempts++;
-            x = Math::random(level.getWidth());
-            y = Math::random(level.getHeight());
-        } while (!isValidPosition(x, y, weapon) && attempts <= MAX_BONUS_ATTEMPTS);
+        std::vector<ValidPosition> validPositionList;
```

Comment:
Na tomhle vectoru nekde zavolej reserve( sirka x vyska levelu ), aby se nemusel nekolikrat resizovat.

A jestli nechces (z duvodu cistoty kodu) predavat ten vector jako parametr do findValidPositions, tak ho muzes z te funkce normalne vratit pokazde novy. Diky copy ellision je to zadarmo.

Reply:
Mrkni ted?